src/libmongoc/src/mongoc/mongoc-collection.h

```diff
@@ -351,6 +351,57 @@ mongoc_collection_estimated_document_count (
    bson_t *reply,
    bson_error_t *error);
 
+
+typedef enum {
+   MONGOC_INDEX_BACKGROUND,
+   MONGOC_INDEX_NAME,
+   MONGOC_INDEX_UNIQUE,
+   MONGOC_INDEX_PARTIAL_FILTER_EXPRESSION,
+   MONGOC_INDEX_SPARSE,
+   MONGOC_INDEX_EXPIRE_AFTER_SECONDS,
+   MONGOC_INDEX_HIDDEN,
+   MONGOC_INDEX_STORAGE_ENGINE,
+   MONGOC_INDEX_WEIGHTS,
+   MONGOC_INDEX_DEFAULT_LANGUAGE,
+   MONGOC_INDEX_LANGUAGE_OVERRIDE,
+   MONGOC_INDEX_TEXT_INDEX_VERSION,
+   MONGOC_INDEX_BITS,
+   MONGOC_INDEX_TWO_D_SPHERE_INDEX_VERSION,
+   MONGOC_INDEX_MIN,
+   MONGOC_INDEX_MAX,
+   MONGOC_INDEX_BUCKET_SIZE,
+   MONGOC_INDEX_COLLATION,
+   MONGOC_INDEX_WILDCARD_PROJECTION,
+} mongoc_set_index_opt_t;
```

Comment:
Users can specify the option they are trying to set with this enum.

Reply:
Though this aligns with APIs like cURL's [curl_easy_setopt](https://curl.haxx.se/libcurl/c/curl_easy_setopt.html) and there are trade-offs, I'd vote for continuing libmongoc's established pattern of representing command options with a `bson_t`, and not having this enum.

An advantage of accepting `bson_t` options is forward compatibility. The implementation will inspect the options for necessary client-side checks (i.e. check that `commitQuorum` is not being used on a pre-4.4 server), but otherwise append them to the command. As options are introduced in the future, we do not need to backport changes to support them (users can just append extra options).